src/commands/execution_ctx.c

```diff
@@ -90,13 +89,25 @@ ExecutionCtx *ExecutionCtx_FromQuery(const char *query) {
 
 	// No cached execution plan, try to parse the query.
 	AST *ast = _ExecutionCtx_ParseAST(query_string, params_parse_result);
-	// If query parsing failed, return an invalid context.
-	if(!ast) return _ExecutionCtx_New(NULL, NULL, EXECUTION_TYPE_INVALID);
+	// If query parsing failed, return NULL.
+	if(!ast) return NULL;
 
 	ExecutionType exec_type = _GetExecutionTypeFromAST(ast);
 	// In case of valid query, create execution plan, and cache it and the AST.
 	if(exec_type == EXECUTION_TYPE_QUERY) {
 		ExecutionPlan *plan = NewExecutionPlan();
+
+		// TODO: there must be a better way to understand if the execution-plan
+		// was constructed correctly,
+		// maybe free the plan within NewExecutionPlan, if error was encountered
+		// and return NULL ?
```

Comment:
Since `NewExecutionPlan` is also invoked by `_ExecutionPlan_UnionPlans`, this would require extra handling there, but we can make those modifications.